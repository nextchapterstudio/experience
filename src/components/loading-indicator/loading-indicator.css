.loading-indicator {
  display: inline-block;
  width: 128px;
  height: 128px;
  position: absolute;
  top: 200px;
  left: 450px;
}

.loading-indicator:after {
  content: ' ';
  display: block;
  width: 46px;
  height: 46px;
  margin: 1px;
  border-radius: 50%;
  border: 5px solid #fff;
  border-color: #fff transparent #fff transparent;
  animation: lds-dual-ring 1.2s linear infinite;
}

@keyframes lds-dual-ring {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}

@media screen and (max-width: 1024px) {
  .loading-indicator {
    display: inline-block;
    width: 128px;
    height: 128px;
    position: absolute;
    top: 200px;
    left: 250px;
  }

  .loading-indicator:after {
    content: ' ';
    display: block;
    width: 46px;
    height: 46px;
    margin: 1px;
    border-radius: 50%;
    border: 5px solid #fff;
    border-color: #fff transparent #fff transparent;
    animation: lds-dual-ring 1.2s linear infinite;
  }
}

@media screen and (max-width: 400px) {
  .loading-indicator {
    display: inline-block;
    width: 128px;
    height: 128px;
    position: absolute;
    top: 200px;
    left: 110px;
  }

  .loading-indicator:after {
    content: ' ';
    display: block;
    width: 46px;
    height: 46px;
    margin: 1px;
    border-radius: 50%;
    border: 5px solid #fff;
    border-color: #fff transparent #fff transparent;
    animation: lds-dual-ring 1.2s linear infinite;
  }
}
