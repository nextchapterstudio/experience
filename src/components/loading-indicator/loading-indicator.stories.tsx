import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'

import { LoadingIndicator } from './loading-indicator'

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Example/LoadingIndicator',
  component: LoadingIndicator,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {},
} as ComponentMeta<typeof LoadingIndicator>

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof LoadingIndicator> = (args) => <LoadingIndicator {...args} />

export const LoadingIndicatorExample = Template.bind({})
// More on args: https://storybook.js.org/docs/react/writing-stories/args
LoadingIndicatorExample.args = {
  children: ' ',
}
