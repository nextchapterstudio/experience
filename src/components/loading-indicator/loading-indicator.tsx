import React, { FC } from 'react'
import classnames from 'classnames'
import { BaseElement, BaseProps } from '../base-element'
import './loading-indicator.css'
export const LoadingIndicator: FC<BaseProps> = ({ className, children, ...props }) => {
  return (
    <BaseElement className={classnames('loader', className)} {...props}>
      {children}
    </BaseElement>
  )
}

LoadingIndicator.defaultProps = {
  as: 'div',
}

export default LoadingIndicator
