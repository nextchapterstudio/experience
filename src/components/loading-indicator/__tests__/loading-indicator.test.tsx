import React from 'react'
import { render, cleanup } from '@testing-library/react'

import { LoadingIndicator } from '../loading-indicator'

describe('The Loader Component', () => {
  afterEach(cleanup)
  it('should render a div with the class loader', () => {
    const { container } = render(<LoadingIndicator> </LoadingIndicator>)
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('loader')
  })
  it('should render a div with the class loader and the className passed in', () => {
    const { container } = render(<LoadingIndicator className="loader-test" />)
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('loader')
    expect(container.firstChild).toHaveClass('loader-test')
  })
})
