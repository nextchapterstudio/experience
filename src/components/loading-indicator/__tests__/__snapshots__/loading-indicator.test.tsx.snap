// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Loader Component should render a div with the class loader 1`] = `
<div
  class="loader"
>
   
</div>
`;

exports[`The Loader Component should render a div with the class loader and the className passed in 1`] = `
<div
  class="loader loader-test"
/>
`;
